src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Frameworks/ITargetFrameworkProvider.cs

```diff
@@ -8,9 +8,9 @@ namespace Microsoft.VisualStudio.ProjectSystem
     internal interface ITargetFrameworkProvider
     {
         /// <summary>
-        /// Parses full tfm or short framework name and returns a corresponding <see cref="TargetFramework"/>
-        /// instance, or <see langword="null" /> if the framework name has an invalid format.
+        /// Returns a <see cref="TargetFramework"/> instance for the target framework moniker
+        /// or <see langword="null" /> if the framework moniker is null or empty.
         /// </summary>
-        TargetFramework? GetTargetFramework(string? shortOrFullName);
+        TargetFramework? GetTargetFramework(string? targetFrameworkMoniker);
```

Comment:
A quick scan of the code suggests that we never actually pass a nullable string to this method. While you're cleaning this up, consider removing the null propagation. Alternatively, this could be annotated with `[return: NotNullIfNotNull("targetFrameworkMoniker")]`.

Reply:
I was wrong. The method can return `null` when an input is non-null (empty string).